discretize/TensorMesh.py

```diff
@@ -594,93 +598,199 @@ def vol(self):
         return self._vol
 
     @property
+    def areaFx(self):
+        """
+        Area of the x-faces
+        """
+        if getattr(self, '_areaFx', None) is None:
+            # Ensure that we are working with column vectors
+            vh = self.h
+            # The number of cell centers in each direction
+            n = self.vnC
+            # Compute areas of cell faces
+            if self.dim == 1:
+                areaFx = np.ones(n[0]+1)
+            elif self.dim == 2:
+                areaFx = np.outer(np.ones(n[0]+1), vh[1])
+            elif self.dim == 3:
+                areaFx = np.outer(
+                    np.ones(n[0]+1), utils.mkvc(np.outer(vh[1], vh[2]))
+                )
+            self._areaFx = utils.mkvc(areaFx)
+        return self._areaFx
+
+    @property
+    def areaFy(self):
+        """
+        Area of the y-faces
+        """
+        if getattr(self, '_areaFy', None) is None:
+            # Ensure that we are working with column vectors
+            vh = self.h
+            # The number of cell centers in each direction
+            n = self.vnC
+            # Compute areas of cell faces
+            if self.dim == 1:
+                raise Exception('1D meshes do not have y-Faces')
+            elif self.dim == 2:
+                areaFy = np.outer(vh[0], np.ones(n[1]+1))
+            elif self.dim == 3:
+                areaFy = np.outer(
+                    vh[0], utils.mkvc(np.outer(np.ones(n[1]+1), vh[2]))
+                )
+            self._areaFy = utils.mkvc(areaFy)
+        return self._areaFy
+
+    @property
+    def areaFz(self):
+        """
+        Area of the z-faces
+        """
+        if getattr(self, '_areaFz', None) is None:
+            # Ensure that we are working with column vectors
+            vh = self.h
+            # The number of cell centers in each direction
+            n = self.vnC
+            # Compute areas of cell faces
+            if self.dim == 1 or self.dim == 2:
+                raise Exception(
+                    '{}D meshes do not have z-Faces'.format(self.dim)
+                )
+            elif self.dim == 3:
+                areaFz = np.outer(
+                    vh[0], utils.mkvc(np.outer(vh[1], np.ones(n[2]+1)))
+                )
+            self._areaFz = utils.mkvc(areaFz)
+        return self._areaFz
+
+    @property
     def area(self):
         """Construct face areas of the 3D model as 1d array."""
-        if getattr(self, '_area', None) is None:
+        if self.dim == 1:
+            return self.areaFx
+        elif self.dim == 2:
+            return np.r_[self.areaFx, self.areaFy]
+        elif self.dim == 3:
+            return np.r_[self.areaFx, self.areaFy, self.areaFz]
```

Comment:
both seem to be fine for vectors
![image](https://cloud.githubusercontent.com/assets/6361812/26666408/957d9ad8-4654-11e7-9ad6-08deb024a5d9.png)